src/hevm/src/EVM/Transaction.hs

```diff
@@ -23,38 +23,60 @@ import qualified Data.Aeson.Types  as JSON
 import qualified Data.ByteString   as BS
 import qualified Data.Map          as Map
 
-data Transaction = Transaction
-  { txData     :: ByteString,
+data AccessListEntry = AccessListEntry {
+  accessAddress :: Addr,
+  accessStorageKeys :: [W256]
+} deriving Show
+
+data Transaction = Transaction {
+    txData     :: ByteString,
     txGasLimit :: W256,
     txGasPrice :: W256,
     txNonce    :: W256,
     txR        :: W256,
     txS        :: W256,
     txToAddr   :: Maybe Addr,
     txV        :: W256,
-    txValue    :: W256
-  } deriving Show
+    txValue    :: W256,
+    txType     :: Maybe W256,
```

Comment:
could also make this an enumeration of known txTypes, it might make things simpler since we wouldn't have to deal with these extra `Maybe`s